doc/sphinx/README.rst

```diff
@@ -0,0 +1,298 @@
+
+          This command is equivalent to :n:`…`.
+
+``.. cmdv::`` :black_nib: A variant of a Coq command.
+    Example::
+
+       .. cmd:: Axiom @ident : @term.
+
+          This command links *term* to the name *ident* as its specification in
+          the global context. The fact asserted by *term* is thus assumed as a
+          postulate.
+
+          .. cmdv:: Parameter @ident : @term.
+
+             This is equivalent to :n:`Axiom @ident : @term`.
+
+``.. exn::`` :black_nib: An error raised by a Coq command or tactic.
+    This commonly appears nested in the ``.. tacn::`` that raises the
+    exception.
+
+    Example::
+
+       .. tacv:: assert @form by @tactic
+
+          This tactic applies :n:`@tactic` to solve the subgoals generated by
+          ``assert``.
+
+          .. exn:: Proof is not complete
+
+             Raised is :n:`@tactic` does not fully solve the goal.
+
+``.. opt::`` :black_nib: A Coq option.
+    Example::
+
+       .. opt:: Nonrecursive Elimination Schemes
+
+          This option controls whether types declared with the keywords
+          :cmd:`Variant` and :cmd:`Record` get an automatic declaration of the
+          induction principles.
+
+``.. prodn::`` :black_nib: Grammar productions.
+    This is useful if you intend to document individual grammar productions.
+    Otherwise, use Sphinx's `production lists
+    <http://www.sphinx-doc.org/en/stable/markup/para.html#directive-productionlist>`_.
+
+``.. tacn::`` :black_nib: A tactic, or a tactic notation.
+    Example::
+
+       .. tacn:: do @num @expr
+
+          :token:`expr` is evaluated to ``v`` which must be a tactic value. …
+
+``.. tacv::`` :black_nib: A variant of a tactic.
+    Example::
+
+       .. tacn:: fail
+
+          This is the always-failing tactic: it does not solve any goal. It is
+          useful for defining other tacticals since it can be caught by
+          :tacn:`try`, :tacn:`repeat`, :tacn:`match goal`, or the branching
+          tacticals. …
+
+          .. tacv:: fail @natural
+
+             The number is the failure level. If no level is specified, it
+             defaults to 0. …
+
+``.. thm::`` A theorem.
+    Example::
+
+       .. thm:: Bound on the ceiling function
+
+          Let :math:`p` be an integer and :math:`c` a rational constant. Then
+          :math:`p \ge c \rightarrow p \ge \lceil{c}\rceil`.
+
+``.. warn::`` :black_nib: An warning raised by a Coq command or tactic..
+    Do not mistake this for ``.. warning::``; this directive is for warning
+    messages produced by Coq.
+
+
+    Example::
+
+       .. warn:: Ambiguous path
+
+         When the coercion `qualid` is added to the inheritance graph, non
+         valid coercion paths are ignored.
+
+Coq directives
+==============
+
+In addition to the objects above, the ``coqrst`` Sphinx plugin defines the following directives:
+
+``.. coqtop::`` A reST directive to describe interactions with Coqtop.
```

Comment:
This paragraph is interesting but either it is incomplete or I was given misleading information. Aren't there some automatic `Abort` also sent at the end of some `coqtop` directives? Or maybe it's only when leaving a block...

Reply:
Hmm. I don't think it ever calls abort, but there's a `reset` setting.

Reply:
Thanks for clarifying this. Indeed, it has been falsely assumed by many people who did the migration work and because of Coq stupid nested lemmas support, it has gone largely unnoticed but we can find instances which demonstrate the problem: https://coq.inria.fr/refman/proof-engine/tactics.html#coq:tacn.decompose (subgoal remaining after `Qed`). Many a further warning in the documentation about abusing nested lemmas would be good.

Reply:
Done.

Reply:
> stupid nested lemmas support,

Well, I find it pretty convenient :smiley: